test/gurka/test_matrix.cc

```diff
@@ -820,3 +820,72 @@ TEST(StandAlone, CostMatrixTrivialRoutes) {
     EXPECT_EQ(matrix.matrix().shapes(0), encoded);
   }
 }
+
+TEST(StandAlone, HGVNoAccessPenalty) {
+  // if hgv_no_penalty is on we should still respect the maxweight restriction on CD
+  // so we should take the next-best hgv=no edge with JK
+  const std::string ascii_map = R"(
+    A-1-------B----C----D----E--2-------F
+                   |    |
+                   J----K
+                   |    |             
+                   |    |
+                   L----M
+           )";
+
+  const gurka::ways ways = {
+      {"AB", {{"highway", "residential"}, {"hgv", "no"}}},
+      {"BC", {{"highway", "residential"}}},
+      {"CD", {{"highway", "residential"}, {"hgv", "no"}, {"maxweight", "3.5"}}},
+      {"DE", {{"highway", "residential"}}},
+      {"EF", {{"highway", "residential"}, {"hgv", "no"}}},
+      {"CJ", {{"highway", "residential"}}},
+      {"JK", {{"highway", "residential"}, {"hgv", "no"}}},
+      {"JLMK", {{"highway", "residential"}}},
+      {"KD", {{"highway", "residential"}}},
+  };
+
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, 100);
+  gurka::map map = gurka::buildtiles(layout, ways, {}, {}, "test/data/hgv_no_access_penalty",
+                                     {{"service_limits.max_timedep_distance_matrix", "50000"}});
+
+  std::unordered_map<std::string, std::string> cost_matrix =
+      {{"/costing_options/truck/hgv_no_access_penalty", "2000"},
+       {"/sources/0/date_time", "2024-03-20T09:00"},
+       {"/prioritize_bidirectional", "1"}};
+  std::unordered_map<std::string, std::string> td_matrix =
+      {{"/costing_options/truck/hgv_no_access_penalty", "2000"},
+       {"/sources/0/date_time", "2024-03-20T09:00"}};
```

Comment:
making a time request makes sure we can call both costmatrix & tdmatrix in a simple loop below